zerver/views/webhooks/updown.py

```diff
@@ -0,0 +1,77 @@
+# Webhooks for external integrations.
+from __future__ import absolute_import
+import re
+from datetime import datetime, timedelta
+from zerver.lib.actions import check_send_message
+from zerver.lib.response import json_success, json_error
+from zerver.decorator import REQ, has_request_variables, api_key_only_webhook_view
+
+
+SUBJECT_TEMPLATE = "{service_url}"
+
+
+class UnsupprotedUpdownEventType(Exception):
+    pass
+
+def send_message_for_event(event, user_profile, client, stream):
+    event_type = get_event_type(event)
+    subject = SUBJECT_TEMPLATE.format(service_url=event['check']['url'])
+    body = EVENT_TYPE_BODY_MAPPER[event_type](event)
+    check_send_message(user_profile, client, 'stream', [stream], subject, body)
```

Comment:
Somewhere we should be doing validation and throwing a KeyError if a field doesn't exist in the input data.